docs/HowToUsePeripheral.md

```diff
@@ -501,9 +516,222 @@ static const struct i2s_ops_s g_i2sops = {
 	.i2s_err_cb_register = i2s_err_cb_register,
 };
 ```
+
 ### I2S Initialization and Device Access
-Board shall export ```struct i2s_dev_s* xxx_i2s_initialize(uint16_t port)``` to initialize a i2s port and get the corresponding device structure.    
+Board shall export ```struct i2s_dev_s* xxx_i2s_initialize(uint16_t port)``` to initialize a i2s port and get the corresponding device structure.
 **NOTE**: Current code in s5j_i2s.c has initialize function as ```struct i2s_dev_s* s5j_i2s_initialize(uint16_t port)```.
 
 This i2s initialize function shall be called with the port number to get the corresponding device structure.
 
+
+## PWM
+
+All structure and function prototypes to be implemented by low-level PWM drivers.
+PWM drivers are provided in the header file [pwm.h](../os/include/tinyara/pwm.h).
+
+
+### PWM device driver initialzation
+
+Every PWM channel must implement some function to initialize the PWM device like below.
+Plese see ```imxrt_pwm_register```.
+
+```c
+FAR struct pwm_lowerhalf_s * imxrt_pwm_initialize(int channel)
+{
+	switch (channel) {
+#ifdef CONFIG_IMXRT_PWM1
+	case 1:
+		lower = &g_ch1dev;
+		break;
+	...
+#endif
+#ifdef CONFIG_IMXRT_PWM2
+	case 9:
+		lower = &g_ch9dev;
+		break;
+#endif
+	...
+	if (imxrt_pwm_init(base, submodule, &pwmconfig) == kStatus_Fail) {
+		return NULL;
+	}
+	//...
+	imxrt_pwm_setup((FAR struct pwm_lowerhalf_s *)lower);
+
+	return (FAR struct pwm_lowerhalf_s *)lower;
+}
+
+static void imxrt_pwm_register(int ch)
+{
+#ifdef CONFIG_PWM
+	struct pwm_lowerhalf_s *pwm_ch;
+	char path[16];
+	int ret;
+
+	/* The last two channels(15, 16) multiplex the PINs with LPUART1,
+	 * so for debugging they are not be set to PWM PINs */
+	pwm_ch = imxrt_pwminitialize(ch);
+	if (!pwm_ch) {
+		dbg("Failed to get imxrt PWM Channel lower half\n");
+		return;
+	}
+
+	/* Register the PWM channel driver at "/dev/pwmx" */
+	snprintf(path, sizeof(path), "/dev/pwm%d", ch);
+	ret = pwm_register(path, pwm_ch);
+	if (ret < 0) {
+		dbg("Imxrt PWM registeration failure: %d\n", ret);
+	}
+#endif
+	return;
+}
+```
+
+PWM device drive must create an instance of ```struct pwm_ops_s``` and returned higher level device driver as show avobe.
+
+Example implementations is present in [imxrt_pwm.c](../os/arch/arm/src/imxrt/imxrt_pwm.c)
+
+And PWM device will control using below 5 functions.
+
+```c
+struct pwm_ops_s {
+	/* This method is called when the driver is opened.  The lower half driver
+	 * should configure and initialize the device so that it is ready for use.
+	 * It should not, however, output pulses until the start method is called.
+	 */
+
+	CODE int (*setup)(FAR struct pwm_lowerhalf_s *dev);
+
+	/* This method is called when the driver is closed.  The lower half driver
```

Comment:
could you remove the white space?
"opened. The"

Reply:
ok.